### PR TITLE
Support for Amazon devices with Google Play sideloaded

### PR DESCRIPTION
**Pull request by Jon-Salmon**
When the google play store is sideloaded onto an amazon device, it isn't good enough to just check for the presence of the app store. Instead, the store used has to be the store the app was installed from.

This adds support for this in cases where both the Google Play store and the Amazon app store are installed.

### CODE REVIEW

File: android/src/main/java/com/dooboolab/flutterinapppurchase/FlutterInappPurchasePlugin.java [hunk 1]
```diff
@@ -46,9 +56,9 @@ public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
 
   @Override
   public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
-    if (isPackageInstalled(context, "com.android.vending")) {
+    if (isAndroid) {
       tearDownChannel();
-    } else if(isPackageInstalled(context, "com.amazon.venezia")) {
+    } else if(isAmazon) {
       tearDownChannel();
     }
```
Comment (hyochan) [inline android/src/main/java/com/dooboolab/flutterinapppurchase/FlutterInappPurchasePlugin.java:63]: Can we make this shorter to `if (isAndroid && isAmazon)`?

File: android/src/main/java/com/dooboolab/flutterinapppurchase/FlutterInappPurchasePlugin.java [hunk 2]
```diff
@@ -75,19 +85,19 @@ public static void registerWith(Registrar registrar) {
 
   @Override
   public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
-    if (isPackageInstalled(context, "com.android.vending")) {
+    if (isAndroid) {
       androidInappPurchasePlugin.setActivity(binding.getActivity());
-    } else if(isPackageInstalled(context, "com.amazon.venezia")) {
+    } else if(isAmazon) {
       amazonInappPurchasePlugin.setActivity(binding.getActivity());
     }
```
Comment (hyochan) [inline android/src/main/java/com/dooboolab/flutterinapppurchase/FlutterInappPurchasePlugin.java:92]: Same here

**Review by hyochan — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/dooboolab/flutterinapppurchase/FlutterInappPurchasePlugin.java (hunk 2 above)
Comment (Jon-Salmon) [reply android/src/main/java/com/dooboolab/flutterinapppurchase/FlutterInappPurchasePlugin.java:90]: No, as the base object is different

Comment (hyochan) [reply android/src/main/java/com/dooboolab/flutterinapppurchase/FlutterInappPurchasePlugin.java:90]: Oops, my bad. Sorry I've missed it.

**Review by hyochan — APPROVED** [review]
LGTM!

**Pull request merged**